src/app/Plans/components/Wizard/MappingForm.tsx

```diff
@@ -63,16 +68,27 @@ const MappingForm: React.FunctionComponent<IMappingFormProps> = ({
 }: IMappingFormProps) => {
   usePausedPollingEffect();
 
+  const nicProfilesQuery = useNicProfilesQuery(sourceProvider);
+  const disksQuery = useDisksQuery(sourceProvider);
+
+  const rhvResourcesLoaded = nicProfilesQuery.isSuccess && disksQuery.isSuccess;
+
   const mappingResourceQueries = useMappingResourceQueries(
     sourceProvider,
     targetProvider,
     mappingType
   );
+
   const { availableSources, availableTargets } = mappingResourceQueries;
 
   const hasInitialized = React.useRef(false);
   React.useEffect(() => {
-    if (!hasInitialized.current && mappingResourceQueries.status === 'success') {
+    if (
+      !hasInitialized.current &&
+      mappingResourceQueries.status === 'success' &&
+      sourceProvider?.type === 'ovirt' &&
```

Comment:
This will prevent the effect from running when you're using a vsphere source provider. Instead we want something like `&& (sourceProvider?.type !== 'ovirt' || rhvResourcesLoaded)`

Reply:
Good catch!